Makefile

```diff
@@ -113,12 +118,26 @@ endif
 	find . -name \*~ -delete
 	find . -name \#\* -delete
 	rm -f bin/crio
+	rm -f bin/crio.cross.*
 	make -C conmon clean
 	make -C pause clean
 	rm -f test/bin2img/bin2img
 	rm -f test/copyimg/copyimg
 	rm -f test/checkseccomp/checkseccomp
 
+local-cross:
+	@$(MAKE) $(CROSS_BUILD_TARGETS); \
+	for file in $(CROSS_BUILD_TARGETS); do test -s $$file && exit 1 ; done
+
+.PHONY: bin/crio.cross.%
+bin/crio.cross.%: .gopathok $(shell hack/find-godeps.sh $(GOPKGDIR) cmd/crio $(PROJECT))
+	@rm -f $@.failed
+	@mkdir -p bin
+	@echo "==> building $@"; \
+	GOOS=`echo "$*" | cut -d . -f 1` \
+	GOARCH=`echo "$*" | cut -d . -f 2` \
```

Comment:
You can skip the `echo` and `cut` and stick to [POSIX shell][1] here.  Here's an example with a dummy target:

```console
$ cat Makefile 
test.%:
        TARGET="$*"; \
        GOOS="$${TARGET%%.*}"; \
        GOARCH="$${TARGET##*.}"; \
        echo "GOOS=$${GOOS} GOARCH=$${GOARCH}"
$ make test.linux.amd64
TARGET="linux.amd64"; \
GOOS="${TARGET%%.*}"; \
GOARCH="${TARGET##*.}"; \
echo "GOOS=${GOOS} GOARCH=${GOARCH}"
GOOS=linux GOARCH=amd64
```

[1]: http://pubs.opengroup.org/onlinepubs/9699919799/utilities/V3_chap02.html#tag_18_06_02

Reply:
cool